src/data/list/join.lean

```diff
@@ -139,4 +142,76 @@ begin
     rw [← drop_take_succ_join_eq_nth_le, ← drop_take_succ_join_eq_nth_le, join_eq, length_eq] }
 end
 
+lemma join_drop_length_sub_one {L : list (list α)} (h : L ≠ []) :
+  (L.drop (L.length - 1)).join = L.last h :=
+begin
+  induction L using list.reverse_rec_on,
+  { cases h rfl },
+  { simp },
+end
+
+section take_join
+
+noncomputable theory
+
+private def get_max [inhabited α] [linear_order α] (P : α → Prop) :=
```

Comment:
I'm pretty suspicious of whether we want this definition, but everything not on this line or below in this file looks like a good set of additions after some small cleanup.

Reply:
Would `take_join_of_lt` be a good addition, too? Should I work on adapting the proof for mathlib?

Reply:
Since nobody was interested in the remaining part, I deleted it from the PR.